controllers/observability_reconciler.go

```diff
@@ -0,0 +1,111 @@
+package controllers
+
+import (
+	"context"
+	"sync"
+
+	kuadrantv1beta1 "github.com/kuadrant/kuadrant-operator/api/v1beta1"
+	"github.com/kuadrant/kuadrant-operator/pkg/utils"
+	"github.com/kuadrant/policy-machinery/controller"
+	"github.com/kuadrant/policy-machinery/machinery"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	monclient "github.com/prometheus-operator/prometheus-operator/pkg/client/versioned"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/utils/ptr"
+)
+
+type ObservabilityReconciler struct {
+	Client     *dynamic.DynamicClient
+	restMapper meta.RESTMapper
+	MonClient  monclient.Interface
+}
+
+func NewObservabilityReconciler(client *dynamic.DynamicClient, rm meta.RESTMapper, mc monclient.Interface) *ObservabilityReconciler {
```

Comment:
nit: when it comes to naming, are we using this initialism as a standard in this code base?

Reply:
do you mean the param names like `rm` for the RESTMapper param?